sema4ai/src/sema4ai_code/robocorp_language_server.py

```diff
@@ -2241,6 +2261,54 @@ def _refresh_agent_spec(self, params: AgentSpecPathDict) -> ActionResultDict:
 
         return ActionResult(success=True, message=None).as_dict()
 
+    def _fix_wrong_agent_import(self, agent_dir, monitor: IMonitor) -> ActionResultDict:
+        import shutil
+        import zipfile
+
+        agent_root_dir = Path(agent_dir)
+        actions_dir: Path = (agent_root_dir / "actions").absolute()
+        had_zips = False
+
+        try:
+            for zip_path in actions_dir.rglob("*.zip"):
+                # Check if a package.yaml exists in the same directory as the zip file then skip
+                if (zip_path.parent / "package.yaml").exists():
+                    continue
+
+                had_zips = True
+                zip_folder_name = zip_path.name.replace(".zip", "")
+                temp_extract_path = zip_path.parent / zip_folder_name
+
+                with zipfile.ZipFile(zip_path, "r") as zip_ref:
+                    zip_ref.extractall(temp_extract_path)
+
+                # Move files and directories from the versioned folder to the root action directory
+                for item in temp_extract_path.rglob("*"):
+                    if item.is_file() and item.name != "metadata.json":
```

Comment:
This isn't the proper name of the json. I don't remember now the exact name, but you should be able to check it when you build the action package in the test (please also fix the test to check for this exclusion).